app/models/katello/glue/pulp/consumer.rb

```diff
@@ -188,7 +188,7 @@ def install_consumer_errata(errata_ids)
       end
 
       def katello_agent_installed?
-        return false if self.is_a? Hypervisor
+        return false if (self.is_a? Hypervisor) || @new_record_before_save
```

Comment:
Maybe we just catch that error then, given this a layer designed to talk to Pulp? That would make it more obvious what is actually happening. Personally, I just find `@new_record_before_save` not obvious as to why I might need to return false and not query `simple_packages`.